app/Repositories/PostRepository.php

```diff
@@ -81,13 +81,17 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
         // Get the action_id either from the payload or the DB.
         if (isset($data['action_id'])) {
             $actionId = $data['action_id'];
+            $action = Action::findOrFail($data['action_id']);
         } else {
-            $type = isset($data['type']) ? $data['type'] : 'photo';
             $action = Action::where([
                 'campaign_id' => $signup->campaign_id,
-                'post_type' => $type,
+                'post_type' => $data['type'],
                 'name' => $data['action'],
-            ])->firstOrFail();
+            ])->first();
+
+            if (! $action) {
+                abort(500, 'No action found.');
```

Comment:
This branch is named `422`... is there a reason to return a 500 instead of a 422?

Also - would it make sense to log this message so we can find specific instances of these errors in Papertrail?

Reply:
We _could_ manually create a validation error here like so:

```php
throw new ValidationException::withMessages(array_fill_keys(['campaign_id', 'post_type', 'name'], 'An action with the given fields does not exist.'));
```

My concern with that is that we'd be breaking our contract of how validation works, since a request might be partially completed (e.g. a signup has been created, and now we're moving on to creating the post). Ideally, all of our validation happens before we make any "actual" changes to the database.

Reply:
> 
This branch is named `422`... is there a reason to return a 500 instead of a 422?

that was a mistake by my part and got lazy to rename!

> 
Also - would it make sense to log this message so we can find specific instances of these errors in Papertrail?

Yes - thank you for the reminder! Added!

Reply:
@DFurnes ah I see what you're saying about how we'd want this before the signup gets created too! Would it be really messy to put the logic in the [Post Request](https://github.com/DoSomething/rogue/blob/master/app/Http/Requests/PostRequest.php#L27)? Or, I actually think maybe the [Post Controller](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/PostsController.php#L110) might be better to do that logic to to the DB query to the `actions` table with those variables?

Reply:
Since this is temporary, I'm good with manually throwing a `ValidationException` here (see code block above). It's probably not worth complicating things too too much for something we'll likely tear out as soon as other systems are updated to send `action_id`.

(But yeah, it's tricky! Ideally, we want all our validation to happen in the `PostRequest`, for example using the [`exists`](https://laravel.com/docs/5.8/validation#rule-exists) operator. Unfortunately, I don't think Laravel has a good way for us to do as complicated of a validation rule as this would be!)

Reply:
Sounds good, thanks for clarifying, @DFurnes ! Updated :)